### PR TITLE
DOC: update README file

### PR DESCRIPTION
**Pull request by rgommers**
- Remove obsolete Codecov badge
- Add link to docs site
- Add link to Tidelift's security vulnerability reporting page
- Remove weird licensing section.

[ci skip]

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -21,11 +18,13 @@ science, and engineering.  It includes modules for statistics, optimization,
 integration, linear algebra, Fourier transforms, signal and image processing,
 ODE solvers, and more.
 
-- **Website (including documentation):** https://www.scipy.org/
+- **Website:** https://www.scipy.org/
+- **Documentation:** http://docs.scipy.org/
```
Comment (mdhaber) [inline README.rst]: https

**Review by mdhaber — APPROVED** [review]
I agree with removing the blurb about the license file, but why remove the link? Not an issue, just curious.

**Comment by rgommers** [comment]
> I agree with removing the blurb about the license file, but why remove the link? Not an issue, just curious.

I just figured it was obvious. The repo says BSD-3, and there a LICENSE file. Adding a new sentence saying "please see the license file" kind of seems pointless to me.

**Comment by mdhaber** [comment]
Ok do you want to change the docs to https or just merge?

**Comment by rgommers** [comment]
> Ok do you want to change the docs to https or just merge?

changed to https and merged. thanks for the review

**Pull request merged**